### PR TITLE
gh-96142: add missing params to `dataclass._DataclassParams`

### PR DESCRIPTION
**Pull request by sobolevn**
I went with explicit `.__dataclass_params__` attribute access instead of `getattr(klass, _PARAMS)` because it felt more safe and concete to me.

I've also considered `kw_only` argument by @da-woods, but it feels like `_DataclassParams` structure must just hold passed arguments without much logic in it. It is up to the caller to inspect `kw_only` on class/field.

* Issue: gh-96142

### CODE REVIEW

**Comment by da-woods** [comment]
> I've also considered `kw_only` argument by @da-woods, but it feels like `_DataclassParams` structure must just hold passed arguments without much logic in it. It is up to the caller to inspect `kw_only` on class/field.

I think "include all the parameters" is a pretty defensible position too.

File: Lib/test/test_dataclasses.py [hunk 1]
```diff
@@ -68,6 +68,17 @@ def test_field_repr(self):
 
         self.assertEqual(repr_output, expected_output)
 
+    def test_dataclass_params_repr(self):
```
Comment (DanielNoord) [inline Lib/test/test_dataclasses.py:71]: Is it worthwhile to do a test based on `for param in inspect.signature(dataclass).parameters` check? So new flags will automatically fail this test?

Comment (sobolevn) [reply Lib/test/test_dataclasses.py:71]: Added!

**Review by DanielNoord — APPROVED** [review]

File: Lib/test/test_dataclasses.py [hunk 2]
```diff
@@ -68,6 +68,26 @@ def test_field_repr(self):
...
```
Comment (ericvsmith) [inline Lib/test/test_dataclasses.py:71]: Can you add a comment here that says that even though this is testing an internal implementation detail, it's testing a feature we want to make sure is correctly implemented for the sake of dataclasses itself?

File: Lib/test/test_dataclasses.py [hunk 3]
```diff
@@ -68,6 +68,26 @@ def test_field_repr(self):
...
+        class Some: pass
+
+        repr_output = repr(Some.__dataclass_params__)
+        expected_output = "_DataclassParams(init=True,repr=True," \
+                          "eq=True,order=False,unsafe_hash=False,frozen=True," \
+                          "match_args=True,kw_only=False," \
+                          "slots=True,weakref_slot=False)"
+        self.assertEqual(repr_output, expected_output)
+
+    def test_dataclass_params_signature(self):
```
Comment (ericvsmith) [inline Lib/test/test_dataclasses.py:85]: And same thing here.

**Review by ericvsmith — CHANGES_REQUESTED** [review]
This is all great. Thanks! With the one nit about a comment, I'll accept this.

**Comment by sobolevn** [comment]
Done!

I have made the requested changes; please review again.

**Review by ericvsmith — APPROVED** [review]

File: Lib/test/test_dataclasses.py [hunk 4]
```diff
@@ -68,6 +68,32 @@ def test_field_repr(self):
 
         self.assertEqual(repr_output, expected_output)
 
+    def test_dataclass_params_repr(self):
+        # Even though this is testing an internal implementation detail,
+        # it's testing a feature we want to make sure is correctly implemented
```
Comment (ericvsmith) [inline Lib/test/test_dataclasses.py]: There's a trailing space on this line that is causing a CI failure.

File: Lib/test/test_dataclasses.py [hunk 5]
```diff
@@ -68,6 +68,32 @@ def test_field_repr(self):
...
+        repr_output = repr(Some.__dataclass_params__)
+        expected_output = "_DataclassParams(init=True,repr=True," \
+                          "eq=True,order=False,unsafe_hash=False,frozen=True," \
+                          "match_args=True,kw_only=False," \
+                          "slots=True,weakref_slot=False)"
+        self.assertEqual(repr_output, expected_output)
+
+    def test_dataclass_params_signature(self):
+        # Even though this is testing an internal implementation detail,
+        # it's testing a feature we want to make sure is correctly implemented
```
Comment (ericvsmith) [inline Lib/test/test_dataclasses.py]: Same thing here with a trailing space.

**Review by ericvsmith — CHANGES_REQUESTED** [review]
There are some trailing spaces that are causing CI failures. I haven't actually checked every line for trailing whitespace, just the ones I quickly found (I'm working on checking out the code, but you can probably check faster than my connection can clone the code).

**Comment by sobolevn** [comment]
Thanks everyone!

**Pull request merged**